service/history/historyEngine.go

```diff
@@ -3199,11 +3199,19 @@ func (e *historyEngineImpl) GenerateLastHistoryReplicationTasks(
 	}
 
 	now := e.shard.GetTimeSource().Now()
-	err = mutableState.GenerateLastHistoryReplicationTasks(now)
+	task, err := mutableState.GenerateLastHistoryReplicationTasks(now)
 	if err != nil {
 		return nil, err
 	}
-	err = context.UpdateWorkflowExecutionAsActive(now)
+
+	err = e.shard.AddTasks(&persistence.AddTasksRequest{
+		ShardID: e.shard.GetShardID(),
```

Comment:
Any reason not to make shard set ShardID itself?

Reply:
probably no good reason. :)